xsd-fu/templates-cpp/OMEXMLModelObject.template

```diff
@@ -916,12 +901,10 @@ ${customUpdatePropertyContent[prop.name]}
         // Element property ${prop.name} which is not complex (has no
         // sub-elements) which occurs more than once
         std::vector<common::xml::dom::Element> ${prop.name}_nodeList(getChildrenByTagName("${prop.name}"));
-        for (std::vector<common::xml::dom::Element>::iterator elem = ${prop.name}_nodeList.begin();
-             elem != ${prop.name}_nodeList.end();
-             ++elem)
+        for (auto& elem : ${prop.name}_nodeList)
           {
-            std::string text(element.getTextContent());
-            ome::compat::shared_ptr<${prop.methodName}> object(${prop.langTypeNS}::create(text, model));
+            std::string text(element.getTextContent()); /// @todo should this be elem?
```

Comment:
I'll add one.  Looking through the Java template, it certainly looks like this should be fixed to use `elem`, though it will need testing to be sure.